### PR TITLE
Add Esapi validator for email and SQL Order By

### PR DESCRIPTION
**Pull request by phuongnq**
### Ticket reference or full description of what's in the PR
https://github.com/craftersoftware/craftercms/issues/417

### CODE REVIEW

File: validation/src/main/resources/esapi/ESAPI.properties [hunk 1]
```diff
@@ -466,9 +466,13 @@ Validator.HTTPURL=^.*$
 Validator.HTTPJSESSIONID=^[A-Z0-9]{10,32}$
 
 Validator.SITEID=^[a-z0-9\-_]*$
+Validator.EMAIL=^([\\w\\d._\\-#])+@([\\w\\d._\\-#]+[.][\\w\\d._\\-#]+)+$
 Validator.USERNAME=^[a-zA-Z0-9_\\.]{1,255}$
 Validator.ALPHANUMERIC=^[a-zA-Z0-9]*$
 
+# Validation of SQL related input
+Validator.SQL_ORDER_BY=^([a-zA-Z_][a-zA-Z_0-9]*,?)+(\\sasc|\\sdesc)?$
```
Comment (jmendeza) [inline validation/src/main/resources/esapi/ESAPI.properties:474]: I believe these sort order (asc|desc part) should be case-insensitive as well
cc: @sumerjabri

**Review by jmendeza — CHANGES_REQUESTED** [review]

File: validation/src/main/java/org/craftercms/commons/validation/annotations/param/EsapiValidationType.java [hunk 2]
```diff
@@ -24,7 +24,9 @@ public enum EsapiValidationType {
     HTTPParameterName("HTTPParameterName"),
     HTTPURI("HTTPURI"),
     SITE_ID("SITEID"),
-    USERNAME("USERNAME");
+    EMAIL("EMAIL"),
```
Comment (jmendeza) [inline validation/src/main/java/org/craftercms/commons/validation/annotations/param/EsapiValidationType.java:27]: @phuongnq I forgot to mention, please also add error messages here:

https://github.com/craftercms/commons/blob/develop/validation/src/main/resources/crafter/commons/validation/errors.properties#L44

File: validation/src/main/resources/esapi/ESAPI.properties (hunk 1 above)
Comment (jmendeza) [inline validation/src/main/resources/esapi/ESAPI.properties:474]: Why is it allowing a comma between the field name and the sort order?

**Comment by sumerjabri** [comment]
@phuongnq please resubmit this. We had to merge it with issues to address a compilation error (we should indicate in PRs if there are PR to PR dependencies).

**Pull request merged**